src/transformers/generation/utils.py

```diff
@@ -1421,6 +1424,11 @@ def generate(
                 )
             generation_config.max_length = generation_config.max_new_tokens + input_ids_length
 
+        # adjust max_length when using `input_embeds` in decoder-only models
+        elif model_input_name == "inputs_embeds" and inputs_tensor.shape[:-1] != input_ids.shape:
+            if not self.config.is_encoder_decoder:
+                generation_config.max_length -= inputs_tensor.shape[1]
```

Comment:
We can reduce the nesting here 

```suggestion
        elif (
            model_input_name == "inputs_embeds" and 
            inputs_tensor.shape[:-1] != input_ids.shape and
            not self.config.is_encoder_decoder
        ):
            generation_config.max_length -= inputs_tensor.shape[1]
```